circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/util/FileSystemPathResolver.java

```diff
@@ -17,24 +17,49 @@
 package com.hotels.bdp.circustrain.avro.util;
 
 import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.springframework.util.StringUtils.isEmpty;
 
+import java.io.IOException;
 import java.net.URI;
+import java.net.URISyntaxException;
+
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hdfs.DFSConfigKeys;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class NameServicePathResolver {
-  private static final Logger LOG = LoggerFactory.getLogger(NameServicePathResolver.class);
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+
+public class FileSystemPathResolver {
+  private static final Logger LOG = LoggerFactory.getLogger(FileSystemPathResolver.class);
 
   private final Configuration configuration;
 
-  public NameServicePathResolver(Configuration configuration) {
+  public FileSystemPathResolver(Configuration configuration) {
     this.configuration = configuration;
   }
 
-  public Path resolve(String url) {
+  public Path resolveScheme(Path path) {
+    return resolveScheme(path.toString());
+  }
+
+  public Path resolveScheme(String url) {
+    try {
+      URI uri = new URI(url);
+      if (isEmpty(uri.getScheme())) {
+        String scheme = FileSystem.get(configuration).getScheme();
```

Comment:
OK, so this is saying that if the scheme is empty, we get a default scheme from the config and use that? And for the setup that raised this issue this turns out to be "hdfs" so it resolves the issue? If so then the comment in the changelog about the new config param needs to be removed as there isn't one anymore.